<a href="https://colab.research.google.com/github/Apolo073/CODER_Cienciadedatos/blob/main/PryectoDS_Parte1_Stein.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DATA SCIENCE CODERHOUSE [#61750]

#### ALUMNO: STEIN NAHUEL

## INTRODUCCIÓN

## DESCRIPCIÓN DEL PROBLEMA

## OBJETIVO

## FUENTE

### IMPORTAR LAS LIBRERIAS

In [2]:
import pandas as pd
import numpy as np
import seaborn as sb
import matplotlib.pyplot as plt
from matplotlib.ticker import PercentFormatter

### CARGAR BASE DE DATOS

In [8]:
url = "https://raw.githubusercontent.com/Apolo073/CODER_Cienciadedatos/refs/heads/main/media%20prediction%20and%20its%20cost.csv"
df = pd.read_csv(url)
df.shape

(60428, 40)

## EXPLORACIÓN DE DATOS

##### En esta sección, exploramos el conjunto de datos para comprender sus características y distribuciones. Esto nos ayudará a identificar patrones y relaciones entre las variables.

Resumen Inicial: Mostrar las primeras filas del DataFrame y obtener una vista previa de los datos.
Estadísticas Descriptivas: Calcular estadísticas descriptivas como media, mediana, desviación estándar, etc.
Distribuciones: Visualizar las distribuciones de las variables numéricas y categóricas.
Gráficos Exploratorios: Crear gráficos como histogramas, diagramas de dispersión, y diagramas de caja para explorar los datos.

### TIPOS DE DATOS

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60428 entries, 0 to 60427
Data columns (total 40 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   food_category               60428 non-null  object 
 1   food_department             60428 non-null  object 
 2   food_family                 60428 non-null  object 
 3   store_sales(in millions)    60428 non-null  float64
 4   store_cost(in millions)     60428 non-null  float64
 5   unit_sales(in millions)     60428 non-null  float64
 6   promotion_name              60428 non-null  object 
 7   sales_country               60428 non-null  object 
 8   marital_status              60428 non-null  object 
 9   gender                      60428 non-null  object 
 10  total_children              60428 non-null  float64
 11  education                   60428 non-null  object 
 12  member_card                 60428 non-null  object 
 13  occupation                  604

El conjunto de datos contiene un total de 60428 (registros) y 40 columnas (variables).

No se observan columnas con valores faltantes (non-null count igual que el número total de filas).

---
Variables Potencialmente Relevantes:

Algunas variables que podrían ser relevantes para predecir ..... incluyen "loan_limit," "loan_type," "loan_purpose," "Credit_Worthiness," "loan_amount," "rate_of_interest," "Income," "Credit_Score," "LTV," y "dtir1."


### DESCRIPCIÓN DE LOS DATOS

In [13]:
df.describe()

store_sales(in millions)  store_cost(in millions)  \
count              60428.000000             60428.000000   
mean                   6.541031                 2.619460   
std                    3.463047                 1.453009   
min                    0.510000                 0.163200   
25%                    3.810000                 1.500000   
50%                    5.940000                 2.385600   
75%                    8.670000                 3.484025   
max                   22.920000                 9.726500   

       unit_sales(in millions)  total_children  avg_cars_at home(approx)  \
count             60428.000000    60428.000000              60428.000000   
mean                  3.093169        2.533875                  2.200271   
std                   0.827677        1.490165                  1.109644   
min                   1.000000        0.000000                  0.000000   
25%                   3.000000        1.000000                  1.000000   
50%                   3.000000        3.000000                  2.000000   
75%                   4.000000        4.000000                  3.000000   
max                   6.000000        5.000000                  4.000000   

       num_children_at_home  avg_cars_at home(approx).1           SRP  \
count          60428.000000                60428.000000  60428.000000   
mean               0.829351                    2.200271      2.115258   
std                1.303424                    1.109644      0.932829   
min                0.000000                    0.000000      0.500000   
25%                0.000000                    1.000000      1.410000   
50%                0.000000                    2.000000      2.130000   
75%                1.000000                    3.000000      2.790000   
max                5.000000                    4.000000      3.980000   

       gross_weight    net_weight  ...    store_sqft  grocery_sqft  \
count  60428.000000  60428.000000  ...  60428.000000  60428.000000   
mean      13.806433     11.796289  ...  27988.477494  19133.799696   
std        4.622693      4.682986  ...   5701.022090   3987.395735   
min        6.000000      3.050000  ...  20319.000000  13305.000000   
25%        9.700000      7.710000  ...  23593.000000  16232.000000   
50%       13.600000     11.600000  ...  27694.000000  18670.000000   
75%       17.700000     16.000000  ...  30797.000000  22123.000000   
max       21.900000     20.800000  ...  39696.000000  30351.000000   

        frozen_sqft     meat_sqft    coffee_bar   video_store     salad_bar  \
count  60428.000000  60428.000000  60428.000000  60428.000000  60428.000000   
mean    5312.852552   3541.846280      0.612646      0.354157      0.587956   
std     1575.907263   1050.471635      0.487150      0.478261      0.492207   
min     2452.000000   1635.000000      0.000000      0.000000      0.000000   
25%     4746.000000   3164.000000      0.000000      0.000000      0.000000   
50%     5062.000000   3375.000000      1.000000      0.000000      1.000000   
75%     5751.000000   3834.000000      1.000000      1.000000      1.000000   
max     9184.000000   6122.000000      1.000000      1.000000      1.000000   

       prepared_food       florist          cost  
count   60428.000000  60428.000000  60428.000000  
mean        0.587956      0.562603     99.262366  
std         0.492207      0.496069     30.011257  
min         0.000000      0.000000     50.790000  
25%         0.000000      0.000000     69.650000  
50%         1.000000      1.000000     98.520000  
75%         1.000000      1.000000    126.620000  
max         1.000000      1.000000    149.750000  

[8 rows x 23 columns]

In [14]:
df.head(10)

food_category food_department food_family  store_sales(in millions)  \
0  Breakfast Foods    Frozen Foods        Food                      7.36   
1  Breakfast Foods    Frozen Foods        Food                      5.52   
2  Breakfast Foods    Frozen Foods        Food                      3.68   
3  Breakfast Foods    Frozen Foods        Food                      3.68   
4  Breakfast Foods    Frozen Foods        Food                      4.08   
5  Breakfast Foods    Frozen Foods        Food                      4.08   
6  Breakfast Foods    Frozen Foods        Food                      5.44   
7  Breakfast Foods    Frozen Foods        Food                      3.74   
8  Breakfast Foods    Frozen Foods        Food                      4.08   
9  Breakfast Foods    Frozen Foods        Food                      9.72   

   store_cost(in millions)  unit_sales(in millions)         promotion_name  \
0                   2.7232                      4.0           Bag Stuffers   
1                   2.5944                      3.0  Cash Register Lottery   
2                   1.3616                      2.0    High Roller Savings   
3                   1.1776                      2.0  Cash Register Lottery   
4                   1.4280                      3.0       Double Down Sale   
5                   1.4688                      3.0       Double Down Sale   
6                   2.5568                      4.0  Cash Register Lottery   
7                   1.6082                      2.0  Cash Register Lottery   
8                   1.4688                      3.0  Cash Register Lottery   
9                   4.5684                      3.0    High Roller Savings   

  sales_country marital_status gender  ...  grocery_sqft frozen_sqft  \
0           USA              M      F  ...       18670.0      5415.0   
1           USA              M      M  ...       18670.0      5415.0   
2           USA              S      F  ...       18670.0      5415.0   
3           USA              M      F  ...       18670.0      5415.0   
4           USA              M      M  ...       18670.0      5415.0   
5           USA              M      F  ...       18670.0      5415.0   
6           USA              S      F  ...       18670.0      5415.0   
7           USA              S      M  ...       18670.0      5415.0   
8           USA              S      F  ...       18670.0      5415.0   
9           USA              S      F  ...       18670.0      5415.0   

  meat_sqft coffee_bar video_store  salad_bar prepared_food  florist  \
0    3610.0        1.0         1.0        1.0           1.0      1.0   
1    3610.0        1.0         1.0        1.0           1.0      1.0   
2    3610.0        1.0         1.0        1.0           1.0      1.0   
3    3610.0        1.0         1.0        1.0           1.0      1.0   
4    3610.0        1.0         1.0        1.0           1.0      1.0   
5    3610.0        1.0         1.0        1.0           1.0      1.0   
6    3610.0        1.0         1.0        1.0           1.0      1.0   
7    3610.0        1.0         1.0        1.0           1.0      1.0   
8    3610.0        1.0         1.0        1.0           1.0      1.0   
9    3610.0        1.0         1.0        1.0           1.0      1.0   

           media_type    cost  
0  Daily Paper, Radio  126.62  
1  Daily Paper, Radio   59.86  
2  Daily Paper, Radio   84.16  
3     In-Store Coupon   95.78  
4               Radio   50.79  
5               Radio   50.79  
6     In-Store Coupon   95.78  
7  Daily Paper, Radio   59.86  
8  Daily Paper, Radio   59.86  
9  Daily Paper, Radio   84.16  

[10 rows x 40 columns]